api/v1beta2/flowcollector_types.go

```diff
@@ -562,74 +562,95 @@ type LokiManualParams struct {
-	//+kubebuilder:default:="http://loki:3100/"
+// LokiMicroservicesParams is the configuration for microservices Loki (https://grafana.com/docs/loki/latest/fundamentals/architecture/deployment-modes/#microservices-mode)
+type LokiMicroservicesParams struct {
+	//+kubebuilder:default:="http://loki-distributor:3100/"
 	// `ingesterUrl` is the address of an existing Loki ingester service to push the flows to.
 	IngesterURL string `json:"ingesterUrl,omitempty"`
 
-	//+kubebuilder:validation:optional
-	// `querierURL` specifies the address of the Loki querier service, in case it is different from the
-	// Loki ingester URL. If empty, the URL value is used (assuming that the Loki ingester
-	// and querier are in the same server).
+	//+kubebuilder:default:="http://loki-query-frontend:3100/"
+	// `querierURL` specifies the address of the Loki querier service.
 	QuerierURL string `json:"querierUrl,omitempty"`
 
+	//+kubebuilder:default:="netobserv"
+	// `tenantID` is the Loki `X-Scope-OrgID` header that identifies the tenant for each request.
+	TenantID string `json:"tenantID,omitempty"`
+
 	// TLS client configuration for Loki URL.
 	// +optional
 	TLS ClientTLS `json:"tls"`
 }
 
-// LokiMonolithParams defines the parameters to connect loki for monolithic mode
+// LokiMonolithParams is the configuration for monolithic Loki (https://grafana.com/docs/loki/latest/fundamentals/architecture/deployment-modes/#monolithic-mode)
 type LokiMonolithParams struct {
 	//+kubebuilder:default:="http://loki:3100/"
-	// `url` is the unique address of an existing Loki service that point both ingester and querier.
+	// `url` is the unique address of an existing Loki service that points to both the ingester and the querier.
 	URL string `json:"url,omitempty"`
 
+	//+kubebuilder:default:="netobserv"
+	// `tenantID` is the Loki `X-Scope-OrgID` header that identifies the tenant for each request.
+	TenantID string `json:"tenantID,omitempty"`
+
 	// TLS client configuration for Loki URL.
 	// +optional
 	TLS ClientTLS `json:"tls"`
 }
 
-// LokiStack defines the name and namespace of the loki-operator instance
-type LokiStack struct {
+// LokiStackRef defines the name and namespace of the LokiStack instance
+type LokiStackRef struct {
+	// Name of an existing LokiStack resource to use.
 	//+kubebuilder:default:="loki"
+	//+kubebuilder:validation:Required
 	Name string `json:"name,omitempty"`
-	//+kubebuilder:default:="netobserv"
+
+	// Namespace where this `LokiStack` resource is located. If omited, it is assumed to be the same as `spec.namespace`.
+	// +optional
 	Namespace string `json:"namespace,omitempty"`
 }
 
+type LokiMode string
+
 const (
-	LokiModeManual      = "MANUAL"
-	LokiModeDistributed = "DISTRIBUTED"
-	LokiModeMonolith    = "MONOLITH"
-	LokiModeLokiStack   = "LOKISTACK"
+	LokiModeManual        LokiMode = "Manual"
+	LokiModeLokiStack     LokiMode = "LokiStack"
+	LokiModeMonolithic    LokiMode = "Monolithic"
+	LokiModeMicroservices LokiMode = "Microservices"
```

Comment:
Why these should be camel case when we use uppercase for `agent type`, `deployment mode`, `HPA status`, `Loki auth`, `SASL`, `Exporter type` ?

Reply:
cf https://github.com/netobserv/network-observability-operator/pull/394#discussion_r1304927133
We did it wrong on the others.
Now we have a mix, we should eventually make them all consistent, but from now on, like we did on Features, we should adopt the correct convention

Reply:
so to avoid having a mix I would suggest to fix everything at once for `v1beta2` since it's already an API change. WDYT ?

Reply:
I agree to do it before releasing v1beta2, but rather on a different PR to not mix it up

Reply:
I created a task: https://issues.redhat.com/browse/NETOBSERV-1374
We should add there everything that needs to be cleaned up before we do the first release, ie. any breaking change that we might still find before 1.5